pkg/net/libp2p/key_test.go

```diff
@@ -136,14 +136,17 @@ func TestNetworkPublicKeyToOperatorPublicKey(t *testing.T) {
 		t.Errorf("operator public key uses the wrong curve")
 	}
 
-	if operatorPublicKey.X.Cmp(
-		networkPublicKey.(*libp2pcrypto.Secp256k1PublicKey).X,
-	) != 0 {
+	publicKey, ok := networkPublicKey.(*libp2pcrypto.Secp256k1PublicKey)
+	if !ok {
+		t.Fatal("network public key is not an instance of the libp2p secp256k1 public key")
+	}
+
+	btcecPublicKey := (*btcec.PublicKey)(publicKey)
```

Comment:
Let's explain this conversion.

Reply:
Done in 58c5230ded309eb47e97dd595427798f6ac0d71d.